src/cdsetool/download.py

```diff
@@ -97,6 +99,10 @@ def download_feature(
                         log.warning(e)
                         continue
                 # Close file before copy so all buffers are flushed.
+                if checksum:
+                    checksum_for_file(tmp_file, title, options.get("proxies", None))
```

Comment:
There is no check of the return value from this function, won't this code continue to retry to download until it runs of out retries?

Reply:
That's true I missed that part 
```python
if checksum:
    if not checksum_for_file(tmp_file, title, options.get("proxies", None)):
        os.remove(tmp_file)
        continue  # download retry
```
this should solve the problem